packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -672,45 +690,10 @@ class _ViewContentState extends State<_ViewContent> {
     result = widget.suggestionsBuilder(context, _controller);
     final Size updatedScreenSize = MediaQuery.of(context).size;
 
-    if (_screenSize != updatedScreenSize) {
+    if (_screenSize != updatedScreenSize && widget.showFullScreenView) {
       _screenSize = updatedScreenSize;
-      setState(() {
-        final Rect anchorRect = widget.getRect() ?? _viewRect;
-        final BoxConstraints constraints = widget.viewConstraints ?? widget.viewTheme.constraints ?? widget.viewDefaults.constraints!;
-        final double viewWidth = clampDouble(anchorRect.width, constraints.minWidth, constraints.maxWidth);
-        final double viewHeight = clampDouble(_screenSize!.height * 2 / 3, constraints.minHeight, constraints.maxHeight);
-        final Size updatedViewSize = Size(viewWidth, viewHeight);
-
-        switch (Directionality.of(context)) {
-          case TextDirection.ltr:
-            final double viewLeftToScreenRight = _screenSize!.width - anchorRect.left;
-            final double viewTopToScreenBottom = _screenSize!.height - anchorRect.top;
-
-            // Make sure the search view doesn't go off the screen when the screen
-            // size is changed. If the search view doesn't fit, move the top-left
-            // corner of the view to fit the window. If the window is smaller than
-            // the view, then we resize the view to fit the window.
-            Offset topLeft = anchorRect.topLeft;
-            if (viewLeftToScreenRight < viewWidth) {
-              topLeft = Offset(_screenSize!.width - math.min(viewWidth, _screenSize!.width), anchorRect.top);
-            }
-            if (viewTopToScreenBottom < viewHeight) {
-              topLeft = Offset(topLeft.dx, _screenSize!.height - math.min(viewHeight, _screenSize!.height));
-            }
-            _viewRect = topLeft & updatedViewSize;
-            return;
-          case TextDirection.rtl:
-            final double viewTopToScreenBottom = _screenSize!.height - anchorRect.top;
-            Offset topLeft = Offset(
-              math.max(anchorRect.right - viewWidth, 0.0),
-              anchorRect.top,
-            );
-            if (viewTopToScreenBottom < viewHeight) {
-              topLeft = Offset(topLeft.dx, _screenSize!.height - math.min(viewHeight, _screenSize!.height));
-            }
-            _viewRect = topLeft & updatedViewSize;
-        }
-      });
+      assert(_screenSize != null);
```

Comment:
It looks like the previous line be 
```dart
_screenSize = updatedScreenSize!;
```

Reply:
Ah yes. Removed the assert check. Thanks!